polygon/heimdall/simulator/heimdall_simulator.go

```diff
@@ -0,0 +1,169 @@
+package simulator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ledgerwatch/log/v3"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/ledgerwatch/erigon-lib/chain/snapcfg"
+	"github.com/ledgerwatch/erigon-lib/common/background"
+	"github.com/ledgerwatch/erigon-lib/downloader/snaptype"
+	"github.com/ledgerwatch/erigon/eth/ethconfig"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+	"github.com/ledgerwatch/erigon/turbo/snapshotsync/freezeblocks"
+)
+
+type HeimdallSimulator struct {
+	ctx                context.Context
+	knownSnapshots     *freezeblocks.RoSnapshots
```

Comment:
can remove knownSnapshots and activeSnapshots - you won't need any of the RoSnapshots for this, only the BorRoSnapshots